test/legacy-utils.ts

```diff
@@ -13,7 +13,7 @@ export function transpileString(
     const { diagnostics, file } = transpileStringResult(str, options);
     expect(file.lua).toBeDefined();
 
-    const errors = diagnostics.filter(d => !ignoreDiagnostics || d.source === "typescript-to-lua");
+    const errors = ignoreDiagnostics ? [] : diagnostics.filter(d => d.source === "typescript-to-lua");
```

Comment:
I moved it out of the filter function because `!ignoreDiagnostics || d.source === "typescript-to-lua"` is not correct. It will not ignore tstl diagnostics, which I thought was the purpose of this flag.
Or is this flag only supposed to ignore TS errors?

Im also fine with using `d.category === error` instead as a temporary workaround. We should try to get rid of legacy utils anyway, I might do a follow up PR for that.

Reply:
like this?

`const errors = diagnostics.filter(d => (!ignoreDiagnostics || d.source === "typescript-to-lua") && d.category === ts.DiagnosticCategory.Error);`

Reply:
yeah that's what I was thinking of